### ISSUE TITLE
Unwrap the cause of exception wrapped by Reflection

### ISSUE THREAD

**Issue by MeSaNei** [sbtqa/page-factory#76]
Hello!

I just started to work with page-factory. 
When I run page-factory-example I get an error:

> FactoryRuntimeException:
> Error while executing action 'проверяет присутствие продукта' on YmSearchResultsPage . See the caused exception below

This caused by:
ru.sbtqa.tag.pagefactoryexample.pages.YandexMarket.YmSearchResultsPage.compareProductCost(YmSearchResultsPage.java:46)
`Assert.fail("Продукт " + productName +  " не был найден");`

From:
ru.sbtqa.tag.pagefactory.Page.executeMethodByTitle(Page.java:856)
at
`} catch (NoSuchMethodException | IllegalAccessException | InvocationTargetException e) {`

The reason is that the reflection layer wraps any exception in an InvocationTargetException(following the case we originally got just new AssertionError(message) from Assert). 

As I said I'm just day 0 with a project and maybe misunderstood something. But I think it will be a good idea just to unwrap the exception in ru.sbtqa.tag.pagefactory.Page.executeMethodByTitle and check the original one and pass it to Allure in case of Assertion.

**Comment by MeSaNei** [comment]
Stack trace:
`                <message>FactoryRuntimeException: Error while executing action 'проверяет присутствие продукта' on YmSearchResultsPage . See the caused exception below</message>
                <stack-trace>ru.sbtqa.tag.pagefactory.exceptions.FactoryRuntimeException: Error while executing action 'проверяет присутствие продукта' on YmSearchResultsPage . See the caused exception below
	at ru.sbtqa.tag.pagefactory.Page.executeMethodByTitle(Page.java:856)
	at ru.sbtqa.tag.pagefactory.stepdefs.GenericStepDefs.userActionOneParam(GenericStepDefs.java:245)
	at ✽.И pagefactory°    пользователь (проверяет присутствие продукта) "KARCHER WD 3 Premium"(YandexMarket.feature:9)
Caused by: java.lang.AssertionError: Продукт KARCHER WD 3 Premium не был найден
	at org.junit.Assert.fail(Assert.java:88)
	at ru.sbtqa.tag.pagefactoryexample.pages.YandexMarket.YmSearchResultsPage.compareProductCost(YmSearchResultsPage.java:46)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.apache.commons.lang3.reflect.MethodUtils.invokeMethod(MethodUtils.java:132)
	at org.apache.commons.lang3.reflect.MethodUtils.invokeMethod(MethodUtils.java:97)
	at ru.sbtqa.tag.pagefactory.Page.executeMethodByTitle(Page.java:853)
	at ru.sbtqa.tag.pagefactory.stepdefs.GenericStepDefs.userActionOneParam(GenericStepDefs.java:245)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at cucumber.runtime.Utils$1.call(Utils.java:40)
	at cucumber.runtime.Timeout.timeout(Timeout.java:16)
	at cucumber.runtime.Utils.invoke(Utils.java:34)
	at cucumber.runtime.java.JavaStepDefinition.execute(JavaStepDefinition.java:38)
	at cucumber.runtime.StepDefinitionMatch.runStep(StepDefinitionMatch.java:37)
	at cucumber.runtime.Runtime.runStep(Runtime.java:300)
	at cucumber.runtime.model.StepContainer.runStep(StepContainer.java:44)
	at cucumber.runtime.model.StepContainer.runSteps(StepContainer.java:39)
	at cucumber.runtime.model.CucumberScenario.run(CucumberScenario.java:44)
	at cucumber.runtime.junit.ExecutionUnitRunner.run(ExecutionUnitRunner.java:102)
	at cucumber.runtime.junit.FeatureRunner.runChild(FeatureRunner.java:63)
	at cucumber.runtime.junit.FeatureRunner.runChild(FeatureRunner.java:18)
	at org.junit.runners.ParentRunner$3.run(ParentRunner.java:290)
	at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:71)
	at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:288)
	at org.junit.runners.ParentRunner.access$000(ParentRunner.java:58)
	at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:268)
	at org.junit.runners.ParentRunner.run(ParentRunner.java:363)
	at cucumber.runtime.junit.FeatureRunner.run(FeatureRunner.java:70)
	at cucumber.api.junit.Cucumber.runChild(Cucumber.java:95)
	at ru.sbtqa.tag.cucumber.TagCucumber.runChild(TagCucumber.java:174)
	at ru.sbtqa.tag.cucumber.TagCucumber.runChild(TagCucumber.java:50)
	at org.junit.runners.ParentRunner$3.run(ParentRunner.java:290)
	at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:71)
	at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:288)
	at org.junit.runners.ParentRunner.access$000(ParentRunner.java:58)
	at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:268)
	at org.junit.runners.ParentRunner.run(ParentRunner.java:363)
	at cucumber.api.junit.Cucumber.run(Cucumber.java:100)
	at org.junit.runner.JUnitCore.run(JUnitCore.java:137)
	at com.intellij.junit4.JUnit4IdeaTestRunner.startRunnerWithArgs(JUnit4IdeaTestRunner.java:68)
	at com.intellij.rt.execution.junit.IdeaTestRunner$Repeater.startRunnerWithArgs(IdeaTestRunner.java:47)
	at com.intellij.rt.execution.junit.JUnitStarter.prepareStreamsAndStart(JUnitStarter.java:242)
	at com.intellij.rt.execution.junit.JUnitStarter.main(JUnitStarter.java:70)
</stack-trace>`

**Comment by kosteman** [comment]
Hello @MeSaNei!
We have wrapped method invoke in executeMethodByTitle because we don't want to enforce users rethrow a lot of exception such as IllegalAccessException, InvocationTargetException in their own code.

But your proposal sounds good and I have created new issue in cucumber-runner project (https://github.com/sbtqa/cucumber-runner/issues/6). We will catch exception on allure level and will unwrap it to caused exception, but before it we need to create pull request to allure-cucumber-jvm project.

I (or one of developers) will take this issue to the next sprint (next week)
Follow this issue.
